notebooks/Cross Validation Primer.ipynb

```diff
@@ -0,0 +1,432 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "# There's No Such Thing as a Free Lunch\n",
+    "\n",
+    "In the field of machine learning, the \"no free lunch\" theorem \"state[s] that any two optimization \n",
+    "algorithms are equivalent when their performance is averaged across all possible problems\" (see [1] \n",
+    "and [2] for some references).That is, we can't tell which machine learning algorithm will best solve the problem in front of us. Of course, with more experience we'll be able to \"intuit\" which ones are better \n",
+    "suited than others, but a priori we can't say which is the best one.\n",
+    "\n",
+    "So we then need a way to measure if machine learning algorithm (A) is better than machine learning \n",
+    "algorithm (B). Or for that matter, if (A) with parameters {l,m,n} or (A) with parameters {p,q,r} will do better. This \n",
+    "is where **cross-validation** comes in.\n",
+    "\n",
+    "[1] Wolpert, David (1996), \"The Lack of A Priori Distinctions between Learning Algorithms\", Neural Computation, pp. 1341-1390\n",
+    "\n",
+    "[2] Wolpert, D.H., and Macready, W.G. (2005) \"Coevolutionary free lunches\", IEEE Transactions on Evolutionary Computation, 9(6): 721-735"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Cross-Validation\n",
+    "\n",
+    "Cross-validation is a way to measure how well a machine learning algorithm will generalize to more data (i.e the \"real\" dataset). For prediction problems, the steps typically go like this:\n",
```

Comment:
In general, the purpose of cross-validation is to simulate the process of going out and collecting more data, without the headache of actually collecting more data. I think that idea is implicit in what you're saying here, but it could be made more explicit.